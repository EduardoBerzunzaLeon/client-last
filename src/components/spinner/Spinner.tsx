import { ProgressSpinner } from 'primereact/progressspinner';
import './spinner.scss';

const Spinner = ({ message }: { message: string }) => (
  <>
    <ProgressSpinner className="flex align-items-center justify-content-center" strokeWidth="4" />
    <span className="flex align-items-center justify-content-center p-spinner-text">{message}</span>
  </>
);

export default Spinner;
