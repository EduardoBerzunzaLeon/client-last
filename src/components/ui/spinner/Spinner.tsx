import { ProgressSpinner } from 'primereact/progressspinner';

interface Props {
  message: string,
  className?: string,
}

export const Spinner = ({ message, className }: Props) => (
  <div className={className}>
    <ProgressSpinner strokeWidth="4" />
    <span className=" font-bold text-2xl text-pink-700">{message}</span>
  </div>

);

Spinner.defaultProps = {
  className: 'flex flex-column align-items-center justify-content-center h-screen',
};

export default Spinner;
