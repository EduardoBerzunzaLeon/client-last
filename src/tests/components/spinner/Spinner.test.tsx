import { shallow } from 'enzyme';
import Spinner from '../../../components/spinner/Spinner';

describe('<Spinner />', () => {
  test('should match to snapshot', () => {
    const wrapper = shallow(<Spinner message="Cargando" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text().trim()).toBe('Cargando');
  });
});
