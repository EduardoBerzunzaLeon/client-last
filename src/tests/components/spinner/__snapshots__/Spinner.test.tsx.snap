// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should match to snapshot 1`] = `
<div
  className="flex flex-column align-items-center justify-content-center h-screen"
>
  <ProgressSpinner
    animationDuration="2s"
    className={null}
    fill="none"
    id={null}
    strokeWidth="4"
    style={null}
  />
  <span
    className=" font-bold text-2xl text-pink-700"
  >
    Cargando
  </span>
</div>
`;
